src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IIsPatternExpression.cs

```diff
@@ -0,0 +1,480 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public partial class IOperationTests : SemanticModelTestBase
+    {
+        [Fact, WorkItem(19927, "https://github.com/dotnet/roslyn/issues/19927")]
+        public void TestIsPatternExpression_VarPatternDeclaration()
+        {
+            string source = @"
+using System;
+class X
+{
+    void M()
+    {
+        int? x = 12;
+        if (/*<bind>*/x is var y/*</bind>*/) Console.WriteLine(y);        
+    }
+}
+";
+            string expectedOperationTree = @"
+IIsPatternExpression (OperationKind.IsPatternExpression, Type: System.Boolean) (Syntax: 'x is var y')
+  Expression: ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: System.Int32?) (Syntax: 'x')
+  Pattern: IDeclarationPattern (Declared Symbol: System.Int32? y) (OperationKind.DeclarationPattern) (Syntax: 'var y')
```

Comment:
>Declared Symbol: System.Int32? y [](start = 32, length = 32)

We should also test what node owns this local. I am suggesting to do this separately, rather than expanding the tested syntax here. We should test that for all IDeclarationPattern nodes.